src/http.cpp

```diff
@@ -684,21 +665,20 @@ m_off_t SpeedController::calculateSpeed(long long numBytes)
 
     while (transferBytes.size())
```

Comment:
You're very right. I didn't pay attention to the cost of the `erase(key)` compared with `erase(pos)`.
```
For the first version (erase(position)), amortized constant.
For the second version (erase(val)), logarithmic in container size.
```